### DISCUSSION TITLE
RAG API Request Not Using Configured Reverse Proxy

### DISCUSSION THREAD

**Discussion by hayschan** [danny-avila/LibreChat#3421] [Q&A]
Hello LibreChat Community,

I'm experiencing an issue with the RAG API setup according to the documentation found here: [RAG API Default Configuration](https://www.librechat.ai/docs/configuration/rag_api#default-configuration). The documentation did not specify related configuration about the reverse proxy settings for RAG API

### Problem Description

When I attempt to upload a file, I receive an "Error processing file" message. Upon inspecting the Docker error logs, it appears that the file text embeddings request is being sent directly to OpenAI's API, not through the configured reverse proxy. Notice that `HTTP Request: POST https://api.openai.com/v1/embeddings`.

The error log indicates a 401 Unauthorized error, suggesting an issue with the API key or its routing.

Here are the relevant log entries:
```python
rag_api           | 2024-07-22 08:41:56,322 - httpx - INFO - HTTP Request: POST https://api.openai.com/v1/embeddings "HTTP/1.1 401 Unauthorized"
rag_api           | 2024-07-22 08:41:56,324 - root - ERROR - Error code: 401 - {'error': {'message': 'Incorrect API key provided: sk-***************XjOj. You can find your API key at https://platform.openai.com/account/api-keys.', 'type': 'invalid_request_error', 'param': None, 'code': 'invalid_api_key'}}
rag_api           | 2024-07-22 08:41:56,325 - root - INFO - Request POST http://rag_api:8000/embed - 200
LibreChat         | 2024-07-22 08:41:56 error: Error embedding file File embedding failed.
LibreChat         | 2024-07-22 08:41:56 error: [/files] Error processing file: File embedding failed.
```

### Suspected Issue

The requests are not being routed through the `OPENAI_REVERSE_PROXY`. Instead, they are being directed to `https://api.openai.com/v1`.

### Current Configuration

Here is an excerpt of the relevant `.env` configuration:
```plaintext
# OpenAI Configuration
OPENAI_API_KEY=API_KEY
OPENAI_REVERSE_PROXY=https://api.gptsapi.net/v1

# RAG API and OpenAI Proxy Setup
RAG_API_URL=http://host.docker.internal:8000
RAG_OPENAI_API_KEY=API_KEY
```

### Questions

1. How can I ensure that the RAG API requests are directed through the configured `OPENAI_REVERSE_PROXY` instead of the default `api.openai.com`?
2. Is there a specific configuration step or setting I might have overlooked that ensures the reverse proxy setting is applied to all outbound API requests?

I would greatly appreciate any advice or insights you can provide to help resolve this issue.

Thank you!

**Comment by khfung** [comment]
Same question 
Thanks in advance

**Comment by gustavw** [comment]
This is what I did to get it working first during the Dockerfile build for RAG i changed to this;

```
# Extend the official LibreChat RAG API image
FROM ghcr.io/danny-avila/librechat-rag-api-dev-lite:latest

# Set working directory
WORKDIR /app

# Update package list and install curl
RUN apt-get update && apt-get install -y curl && rm -rf /var/lib/apt/lists/*

# Enable RAG reverse proxy through downloading tiktoken
ARG ENCODING_MODEL=cl100k_base
# change this to force a re-download of the encoding
ARG ENCODING_CACHE_BUST=1
# don't use /tmp, because a cloud provider may mount ephemeral storage there,
# overwriting our cache
ENV TIKTOKEN_CACHE_DIR=/usr/local/share/tiktoken
ENV TIKTOKEN_ENCODING_URL=https://openaipublic.blob.core.windows.net/encodings/${ENCODING_MODEL}.tiktoken
RUN mkdir -p $TIKTOKEN_CACHE_DIR && \
    curl -L $TIKTOKEN_ENCODING_URL?cache=$ENCODING_CACHE_BUST > \
    $TIKTOKEN_CACHE_DIR/$(echo -n $TIKTOKEN_ENCODING_URL | sha1sum | awk '{ print $1 }')


CMD ["python", "main.py"]
```

I also was strict setting these .env variables. See my example
RAG_API_URL=http://rag.dev-local:8000
RAG_OPENAI_BASEURL=http://nginx.dev-local/openai/v1
RAG_OPENAI_PROXY=http://nginx.dev-local/openai

Hope it helps someone :)
